### DISCUSSION TITLE
Map arbitrary bin widths to same width

### DISCUSSION THREAD

**Discussion by maxgalli** [scikit-hep/mplhep#315] [Q&A]
Hi, I'm opening this thread because I wasn't able to find an easy way to achieve what I'm trying to do.
Let's say I have the following simple reproducer:

```
import hist
from hist import Hist
import matplotlib.pyplot as plt
import mplhep

vals = [40, 38, 39, 43, 11, 14]
edges = [0.0, 0.15, 0.3, 0.6, 0.9, 1.2, 2.5]
axh = hist.axis.Variable(name="my_var", edges=edges)
h = Hist(axh)

h[...] = vals

fig, ax = plt.subplots()
mplhep.histplot(h, yerr=False, histtype="step", binticks=True, ax=ax);
```

This works fine, but what I would like to have is for the edges in ```edges``` to be shown at the same distance, so that it it looks like the bins all have the same width (even if in practice they don't). Pretty much it is the same question asked [here](https://stackoverflow.com/questions/33497559/display-a-histogram-with-very-non-uniform-bin-widths).
Is there an easy way to do this e.g. from the options in ```mplhep.histplot```?

Thanks,

Massimiliano

**Comment by andrzejnovak** [comment]
That's an interesting ask. You want the "plotted" bins to be artificial, but labelled correctly right?

I think that is a bit too niche to implement in mplhep, but the way to go (similar to the SO answer I think) is to redefine the bins on the hist object (or extract the vals and pass arbitrary bins) and then manually set the ticks. What we can improve is that currently passing `bins` along with a hist object gives `TypeError: Cannot give bins with existing histogram` but could probably overwrite the object bins if same length.

**Reply by maxgalli** [reply]
Yes it makes sense, probably the best way is simply redefining the bins in hist and setting the ticks manually. Thanks!
